pkg/brew/brew.go

```diff
@@ -0,0 +1,116 @@
+package brew
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/log"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	brewNameRegex = regexp.MustCompile(`^*.rb$`)
+	//										 $1			 $2	 $3	$4	  $5	  $6	   $7
+	versionRegex = regexp.MustCompile(`^(.*version) (\")((\d+\.)?(\d+\.)?(\*|\d+))(\")$`)
+	//									 $1			$2	$3		   $4
+	shaRegex = regexp.MustCompile(`^(.*sha256) (\")([0-9a-z]+)(\")$`)
+)
+
+//UpdateVersion scans the directory structure rooted in dir for files that match brewNameRegex and replaces any
+// lines starting with FROM <name>:, ENV <name> or ARG=<name> with the newVersion
+func UpdateVersion(dir string, newVersion string) ([]string, error) {
+	//shaPrefix := fmt.Sprintf("url %s:", name)
+	oldVersions := make(map[string]bool)
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```

Comment:
It would have probably been easier to use a `filepath.Glob` here to find the files, as you are just searching by extension.